hudhook/src/hooks/dx11.rs

```diff
@@ -223,37 +143,14 @@ impl ImguiRenderer {
         trace!("Initializing imgui context");
         let imgui_ctx = engine.ctx();
         imgui_ctx.set_ini_filename(None);
-        let mut io = imgui_ctx.io_mut();
-        io.nav_active = true;
-        io.nav_visible = true;
-
-        // Initialize keys
-        io[Key::Tab] = VK_TAB.0 as _;
-        io[Key::LeftArrow] = VK_LEFT.0 as _;
-        io[Key::RightArrow] = VK_RIGHT.0 as _;
-        io[Key::UpArrow] = VK_UP.0 as _;
-        io[Key::DownArrow] = VK_DOWN.0 as _;
-        io[Key::PageUp] = VK_PRIOR.0 as _;
-        io[Key::PageDown] = VK_NEXT.0 as _;
-        io[Key::Home] = VK_HOME.0 as _;
-        io[Key::End] = VK_END.0 as _;
-        io[Key::Insert] = VK_INSERT.0 as _;
-        io[Key::Delete] = VK_DELETE.0 as _;
-        io[Key::Backspace] = VK_BACK.0 as _;
-        io[Key::Space] = VK_SPACE.0 as _;
-        io[Key::Enter] = VK_RETURN.0 as _;
-        io[Key::Escape] = VK_ESCAPE.0 as _;
-        io[Key::A] = VK_A.0 as _;
-        io[Key::C] = VK_C.0 as _;
-        io[Key::V] = VK_V.0 as _;
-        io[Key::X] = VK_X.0 as _;
-        io[Key::Y] = VK_Y.0 as _;
-        io[Key::Z] = VK_Z.0 as _;
-
         let flags = ImguiRenderLoopFlags { focused: true };
 
         trace!("Renderer initialized");
-        ImguiRenderer { engine, render_loop, wnd_proc, flags, swap_chain }
+        let mut renderer = ImguiRenderer { engine, render_loop, wnd_proc, flags, swap_chain };
+
+        ImguiRendererCommon::init_io(&mut renderer);
```

Comment:
I like this. Invoking the trait method as an associated function makes things clear.